### ISSUE TITLE
Figure 3 in the paper is very confusing

### ISSUE THREAD

**Issue by Matsuri247** [Jiawei-Yang/Denoising-ViT#8]
![Snipaste_2024-05-15_12-03-22](https://github.com/Jiawei-Yang/Denoising-ViT/assets/56073360/e3b0c100-2b8e-4606-8faf-6debd6402776)
Thank you for your nice work. However, it seems to be hard to understand the flowchart of DVT. In my point of view, the arrows in fig.3 confuse me a lot. Specifically:
1. For the leftside red box, what happens between the noisy feature maps outputed by ViT and the 3 terms? The arrows make me feel like that the part below is predicting noise in the noisy feature maps. However, i did not find corresponding contents in the paragraphs of the paper.
2. For the rightside red box, the big purple arrows seem to denote that the output of Generalizable Denoiser and the noisy ViT feature maps are the inputs of Per-image denoising. But that seems not to be exactly what fig.(3) express.
In short, it is difficult for me to figure out how exactly DVT inplemented.  I will appreciate it if you offer me help. Looking forward to your reply!

**Comment by Jiawei-Yang** [comment]
Hi, thanks for your interest in our work!

1. In short, the greenish arrows refer to the 'g' in Eq. (5), the orangeish arrows refer to 'f(x)', and the pinkish arrows refer to 'h'. These are also explained in the text right below Figure 3. What might be confusing is the part about the residual predictor. It indeed takes in the original feature map and predicts the residual — the part co-dependent on features and locations, or say, the part that cannot easily be explained away by using `f` and `g`. Feel free to ask for more specific details about any term that you find confusing.

2. The two large triangles (not necessarily arrows) denote an hourglass shape. Imagine the "encoder" and "decoder" architecture in an autoencoder, usually denoted as hourglass architecture as well. The top triangle indicates that we're feeding everything into the per-image denoising module, and the bottom triangle represents the output from that module. The reason we plot arrows pointing towards the outputs of the per-image denoising from the generalizable denoiser is to show that we ask the model to predict the denoised features from the noisy features.

And thanks for your suggestions, we will keep these in mind when we update the paper to make it clearer!

**Comment by Matsuri247** [comment]
> Hi, thanks for your interest in our work!
> 
> 1. In short, the greenish arrows refer to the 'g' in Eq. (5), the orangeish arrows refer to 'f(x)', and the pinkish arrows refer to 'h'. These are also explained in the text right below Figure 3. What might be confusing is the part about the residual predictor. It indeed takes in the original feature map and predicts the residual — the part co-dependent on features and locations, or say, the part that cannot easily be explained away by using `f` and `g`. Feel free to ask for more specific details about any term that you find confusing.
> 2. The two large triangles (not necessarily arrows) denote an hourglass shape. Imagine the "encoder" and "decoder" architecture in an autoencoder, usually denoted as hourglass architecture as well. The top triangle indicates that we're feeding everything into the per-image denoising module, and the bottom triangle represents the output from that module. The reason we plot arrows pointing towards the outputs of the per-image denoising from the generalizable denoiser is to show that we ask the model to predict the denoised features from the noisy features.
> 
> And thanks for your suggestions, we will keep these in mind when we update the paper to make it clearer!

Thank you for your kindly reply! Actually i do notice the text right below, but what i expect to see, that some explainations about the flowchart of how DVT actually works(In fact i am still very confused about it for now...), seem to be missing. I would like to share my current understanding of DVT, and hope you can give some instructions:
1. Should i be very familiar with InstantNGP(mentioned in the supplementary material, but i just know ViT) so that i can comprehend how and why f and g are generated in DVT? It is too suffering to attempt to figure out what everything goes on just from fig.3 and corresponding texts in the paper.
2. In fig.3 (a) Per-image Denoising, I can understand that there is a training process that trying to making the sum of the three terms to be as similar as the noisy ViT feature maps (denoted as _learning the decomposition_). But what's the relationship between this and Per-image Denosing? Do we get noisy-free feature map crops from well-trained Neural Feature Field? 
3. What is the relationship between Per-image Denoising and Generalizable Denoiser? Is Generalizable Denoiser happen after Per-image Denoising?
Besides, it is also extremely difficult for me to read the codes since the code style distincts drastically from what i'm familiar with. I hope you can bear with me if some questions seem silly. Thank you for your patience.
